src/twister2/specification_processor.py

```diff
@@ -235,7 +279,14 @@ def should_skip_for_tag(test_spec: YamlTestSpecification, platform: PlatformSpec
     return False
 
 
-def should_skip_for_arch(test_spec: YamlTestSpecification, platform: PlatformSpecification) -> bool:
+def should_skip_for_arch(
+    test_spec: YamlTestSpecification,
+    platform: PlatformSpecification,
+    arch_from_cli: list[str] = []
```

Comment:
Using list as a default value for function argument is not recommended in Python. In this particular case it doesn't cause any problem, but just to avoid some problems in the future, please remove completely this default value or add mechanism like this:
```python
def should_skip_for_arch(
    test_spec: YamlTestSpecification,
    platform: PlatformSpecification,
    arch_from_cli: None | list[str] = None
) -> bool:
    if arch_from_cli is None:
        arch_from_cli = []
```
You can read more about potential problems with list as a default value here:
https://docs.python-guide.org/writing/gotchas/#mutable-default-arguments 
or here:
https://florimond.dev/en/posts/2018/08/python-mutable-defaults-are-the-source-of-all-evil/